osquery/tables/system/authorized_keys.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+#include <vector>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+#include <osquery/sql.h>
+
+namespace osquery {
+namespace tables {
+
+const std::vector<std::string> kSSHAuthorizedkeys = {
+    ".ssh/authorized_keys",".ssh/authorized_keys2"
+};
+
+void genSSHkeysForUser(const std::string& username,
+                            const std::string& directory,
+                            QueryData& results) {
+  for (const auto& kfile : kSSHAuthorizedkeys) {
+    boost::filesystem::path keys_file = directory;
+    keys_file /= kfile;
+
+    std::string keys_content;
+    if (!readFile(keys_file, keys_content).ok()) {
+      // Cannot read a specific keys file.
+      continue;
+    }
+    //Protocol 1 public key consist of: options, bits, exponent, modulus, comment
+    //Protocol 2 public key consist of: options, keytype, base64-encoded key, comment.
+    for (const auto& line : split(keys_content, "\n")) {
+      if (line[0] != '#') {
```

Comment:
Problem here! Check to make sure the line is not empty: 
```cpp
if (!line.empty() && line[0] !+ '#') {
```

`osquery::split` will trim the line automatically, but you may get empty entries. ;)